WordPress/Classes/Utility/ImageURLs/WPImageURLHelper+Avatars.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  WPImageURLHelper+Avatars.swift
+//  WordPress
+//
+//  Created by Andrew McKnight on 11/22/16.
+//  Copyright © 2016 WordPress. All rights reserved.
+//
+
+import Foundation
+
+// MARK: {Gr|Bl}avatar URLs
+
+extension WPImageURLHelper
+{
+    public class func avatarURL(withHash hash: String, type: WPAvatarSourceType, size: CGSize) -> NSURL? {
```

Comment:
Maybe this can be divided in two different methods, one for the path and one for the components, will make it easier to follow the logic of what is happening.